### ISSUE TITLE
Non optimal resolution

### ISSUE THREAD

**Issue by claes** [kodi-svtplay/xbmc-svtplay#242]
# Problem description
Streaming videos default to a low resolution. 

# Steps to reproduce

Stream any video
Get poor resolution (512×288)

I get higher resolution if I choose Inställningar - Välj program - 6 variant_bitrate ... 
But I then need to do this every time.

I have tried adjusting the setting Bandbredd but any manual value fails to play at all.

 System information
Kodi version: 
18.3 x64
Platform: 
OpenSuse Tumbleweed
Plugin version: 
5.0.1
# Log

**Comment by linqcan** [comment]
Thank you for taking time reporting this issue!

Some background; the addon fetches a tv shows video data URL and gives it to Kodi, which uses ffmpeg (a video library) to decode and play back the video to the user. The addon merely "scrapes" for video URLs. The video data contains information about different versions available related to bandwidth and ffmpeg uses some (unknown to me) magic to determine what version it should use.

Given that, there is nothing really related to addon that would cause your resolution to be lower than expected.

As you noticed there was a bandwidth parser in the addon but it was obsolete (as you also noticed) and was hence removed in version 5.0.2 (that was pushed just hours after this issue was created).

In any case, I could have a look if you could provide a debug log for me?
https://kodi.wiki/view/Log_file/Easy (upload to https://paste.kodi.tv)

**Comment by claes** [comment]
Hi, log uploaded to https://paste.kodi.tv/kawurazotu.kodi
This is now version 5.0.2

**Comment by claes** [comment]
It might be that I captured the log a little differently than described in that link. I reset the log prior to starting the stream, and ended it as it has started. But I think it should have captured what counts.

**Comment by linqcan** [comment]
Thank you for the log file!

I will look at it as soon as possible.

I the meanwhile, I am curious: has this always been the case for you or did
it start happening recently?

**Comment by claes** [comment]
It started after upgrading to Kodi 18.3  + other changes. I used to run Kodi 17.6 (?)  on OpenSuse Tumbleweed and the Packman build of Kodi. This was probably a 6 month or so old installation I upgraded using zypper dup, which updated everything to the latest, incl Kodi 18.3. But then videos did not play at all. I then reinstalled the ffmpeg package to Packman build and got videos to work, but with this problem. Probably OpenSuse had somehow removed codecs from their version of ffmpeg.
So it is likely that ffmpeg has changed its behaviour, among other things.

**Comment by linqcan** [comment]
OK, yeah it does sound like your installation is corrupt somehow.

For reference I tried the same stream on Kodi 18.4 Windows 10 and by comparing logs I found that this trace tells what stream was chosen (still unclear why):
```
DEBUG: CDVDDemuxFFmpeg::AddStream ID: 0
DEBUG: CDVDDemuxFFmpeg::AddStream ID: 7
NOTICE: Opening stream: 7 source: 256
```
`stream: 7` in this case is the high quality version ("Program 6" in the ffmpeg-hls logs).

Maybe a reinstall could solve your issues, otherwise you have start asking around in Kodi and/or ffmpeg forums about why/how the version is chosen.

Sorry I couldn't solve it, I hope this gets you somewhere however!
